### PR TITLE
[Content] Ported Omni Translators (from Polaris), Adds NO_TRANSLATE to Opifex and Cultcommon

### PR DESCRIPTION
**Pull request by IrkallaEpsilon**
https://github.com/PolarisSS13/Polaris/pull/3077 (Anewbe)

which got it from: 

https://github.com/VOREStation/VOREStation/pull/1148 (Arohka)

Thanks to those 2 gigachads from I think 7 years ago theres no need to learn circuits to have a working star trek style omni translator. Hurrah.

## About The Pull Request

Thanks @Dimasw99 for pointing at microphone, TTS and Universal Recorder code. Taking universal recorder code allows this to work without too much effort.

This PR adds and ports the Omni Translator (Handheld and Headset) from Virgo and Polaris. Some minor adjustments to our code needed to be made so it actually works as we do not have listening_object defined. Instead this makes use of bits from the universal recorder, mainly the listening component and the add_hearing component.

For now I added it to SI to one of the bluespace/tcomms technodes since its kinda a communication tool. Also omni translators probably use an advanced algorithm to translate, failing at that use some scifi bullshit to translate. If wanted a mapper can add a few of these pieces to Marshall, Blackshield and maybe council meeting room. The translator (handheld) is reasonably cheap to build. The earpiece needs a bit of silver but less basic mats.

As for Icons: 

If somebody wants to sprite more iconstates for this (so it has an on state) be my guest, I gladly repath the entire thing then and have different icon states for on and off state and if it receives a message and translates it.

Current icons are from Virgo (Arohka).

Tested locally:

Hivemind languages do not work, Jive does not work, obviously this is intended. 

Opifex Language and Cultcommon are not translatable either.

Other languages are translateable to the users choosen output language. 

Tested with a "EUROLANG" speaking Prime, works.

image

image

## Changelog
:cl:
add: Adds universal translators, for now printable by SI. A mapper could add a few to Marshalls, Blackshield and Council Meeting Room probably. The handheld version can also be found on SI computer part disk, SI factory disk and security misc disk as well as be taken for 1 (one) point from loadout tools. PRESS Z TO ACTIVATE THE TRANSLATOR!
/:cl:

### CODE REVIEW

File: code/game/objects/items/devices/translators.dm [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+	item_state = "translator"
+	w_class = ITEM_SIZE_SMALL
+	origin_tech = list(TECH_DATA = 8)
+	matter = list(MATERIAL_STEEL = 4, MATERIAL_GLASS = 4, MATERIAL_PLASTIC = 2)
+	var/mult_icons = 1	//Changes sprite when it translates
+	var/visual = 1		//If you need to see to get the message
+	var/audio = 0		//If you need to hear to get the message
+	var/listening = 0
+	var/datum/language/langset
+
```
Comment (Dimasw99) [inline code/game/objects/items/devices/translators.dm:16]:
```suggestion

/obj/item/device/universal_translator/Destroy()
	if(listening)
		remove_hearing()
	. = ..()

```

**Comment by cdb-is-not-good** [comment]
My only concerns with this are that it needs to be ensured that it can't translate any languages that AI/borgs can't understand as those languages were kept untranslatable for a reason. P sure it's just bird speak tho.

**Comment by cdb-is-not-good** [comment]
I'll leave it at your discretion which of the two solutions is best. If either is implemented I'm perfectly happy.

**Comment by Trilbyspaceclone** [comment]
Throw these into some venders + maybe the hand one in loadout
A good vender would be the toy vender or the marshal one
Also sec-disk/misc would be a good one

**Pull request merged**